--- a/XenosRecomp/shader_common.h
+++ b/XenosRecomp/shader_common.h
@@ -152,11 +152,11 @@
 float3 conanPwlGammaToLinear(float3 g)
 {
     g = saturate(g);
-    float3 hi = select(g >= 192.0 / 255.0, 8.0 / 1024.0, 4.0 / 1024.0);
-    float3 lo = select(g >= 64.0 / 255.0, 2.0 / 1024.0, 1.0 / 1024.0);
-    float3 scale = select(g >= 96.0 / 255.0, hi, lo);
-    float3 offset = select(g >= 96.0 / 255.0, select(g >= 192.0 / 255.0, -1024.0, -256.0),
-                           select(g >= 64.0 / 255.0, -64.0, 0.0));
+    float3 hi = select(g >= 192.0f / 255.0f, 8.0f / 1024.0f, 4.0f / 1024.0f);
+    float3 lo = select(g >= 64.0f / 255.0f, 2.0f / 1024.0f, 1.0f / 1024.0f);
+    float3 scale = select(g >= 96.0f / 255.0f, hi, lo);
+    float3 offset = select(g >= 96.0f / 255.0f, select(g >= 192.0f / 255.0f, -1024.0f, -256.0f),
+                           select(g >= 64.0f / 255.0f, -64.0f, 0.0f));
     float3 l = g * ((255.0 * 1024.0) * scale) + offset;
     l += trunc(l * scale);
     return l * (1.0 / 1023.0);
@@ -209,7 +209,7 @@
 float2 getWeights2D(uint resourceDescriptorIndex, uint samplerDescriptorIndex, float2 texCoord, float2 offset)
 {
     Texture2D<float4> texture = g_Texture2DDescriptorHeap[TEX_INDEX(resourceDescriptorIndex)];
-    return select(isnan(texCoord), 0.0, frac(texCoord * getGuestTexture2DDimensions(resourceDescriptorIndex, texture) + offset - 0.5));
+    return select(isnan(texCoord), 0.0f, frac(texCoord * getGuestTexture2DDimensions(resourceDescriptorIndex, texture) + offset - 0.5));
 }
 
 #ifdef REBLUE_RECOMP
